### PR TITLE
Feature/user model

### PR DESCRIPTION
**Pull request by DanielAugustoUruena**
## Summary

- User model
- Migration with user table

## Known Issues

None.

## Trello Card

[Modelo de usuario](https://trello.com/c/0fuizmX8/21-modelo-de-usuario)

### CODE REVIEW

File: app/models/user.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+module.exports = (sequelize, DataTypes) => {
+  const user = sequelize.define('user', {
+    id: {
+      type: DataTypes.INTEGER,
+      unique: true,
+      autoIncrement: true,
+      primaryKey: true,
+      allowNull: false
+    },
+    first_name: {
```
Comment (jonnyalexbh) [inline app/models/user.js:10]: we use camelCase for models

Comment (jonnyalexbh) [inline app/models/user.js]: we use camelCase for attributes

File: app/models/user.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      type: DataTypes.STRING,
+      allowNull: false
+    },
+    last_name: {
```
Comment (jonnyalexbh) [inline app/models/user.js]: same here

File: migrations/migrations/20210406205811-name-of-migration.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      });
+  },
+
+  down: (queryInterface, Sequelize) => {
+    /*
+      Add reverting commands here.
+      Return a promise to correctly handle asynchronicity.
+
+      Example:
+      return queryInterface.dropTable('users');
```
Comment (jonnyalexbh) [inline migrations/migrations/20210406205811-name-of-migration.js:56]: rollback is not working

File: app/models/user.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      allowNull: false
+    },
+    email: {
+      type: DataTypes.STRING,
+      unique: true,
+      allowNull: false
+    },
+    password: {
+      type: DataTypes.STRING,
+      allowNull: false
```
Comment (jonnyalexbh) [inline app/models/user.js]: think if it is important to use the timestamp fields?

File: app/models/user.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+module.exports = (sequelize, DataTypes) => {
+  const user = sequelize.define('user', {
```
Comment (jonnyalexbh) [inline app/models/user.js]: we recommend calling it User, something like `const User = sequelize.define('User', {`

File: migrations/migrations/20210406205811-name-of-migration.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        },
+        email: {
+          type: Sequelize.STRING,
+          unique: true,
+          allowNull: false
+        },
+        password: {
+          type: Sequelize.STRING,
+          allowNull: false
+        }
```
Comment (jonnyalexbh) [inline migrations/migrations/20210406205811-name-of-migration.js]: if timestamps fields are necessary add them here too

File: app/models/user.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+      unique: true,
+      allowNull: false
+    },
+    password: {
+      type: DataTypes.STRING,
+      allowNull: false
+    }
+  });
+
+  return user;
```
Comment (jonnyalexbh) [inline app/models/user.js]: return `User`

File: app/models/user.js (hunk 1 above)
Comment (DanielAugustoUruena) [reply app/models/user.js]: field name changed to camel case

File: app/models/user.js (hunk 2 above)
Comment (DanielAugustoUruena) [reply app/models/user.js]: field name changed to camel case

File: app/models/user.js (hunk 4 above)
Comment (DanielAugustoUruena) [reply app/models/user.js]: timestamps fields added to model

File: migrations/migrations/20210406205811-name-of-migration.js (hunk 3 above)
Comment (DanielAugustoUruena) [reply migrations/migrations/20210406205811-name-of-migration.js:56]: rollback added

File: app/models/user.js (hunk 5 above)
Comment (DanielAugustoUruena) [reply app/models/user.js]: model name fixed

File: migrations/migrations/20210406205811-name-of-migration.js (hunk 6 above)
Comment (DanielAugustoUruena) [reply migrations/migrations/20210406205811-name-of-migration.js]: timestamp fields adde to table definition

File: app/models/user.js (hunk 7 above)
Comment (DanielAugustoUruena) [reply app/models/user.js]: model name fixed

File: migrations/migrations/20210406205811-name-of-migration.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    /*
+      Add altering commands here.
+      Return a promise to correctly handle asynchronicity.
+
+      Example:
+      return queryInterface.createTable('users', { id: Sequelize.INTEGER });
+    */
+      return queryInterface.createTable('user', {
```
Comment (jonnyalexbh) [inline migrations/migrations/20210406205811-name-of-migration.js]: table names must be plural `users`

File: migrations/migrations/20210406205811-name-of-migration.js [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+  down: (queryInterface, Sequelize) => {
+    /*
+      Add reverting commands here.
+      Return a promise to correctly handle asynchronicity.
+
+      Example:
+      return queryInterface.dropTable('users');
+    */
+      return queryInterface.dropTable('user');
```
Comment (jonnyalexbh) [inline migrations/migrations/20210406205811-name-of-migration.js]: same here `users`

File: migrations/migrations/20210406205811-name-of-migration.js [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+          allowNull: false
+        },
+        email: {
+          type: Sequelize.STRING,
+          unique: true,
+          allowNull: false
+        },
+        password: {
+          type: Sequelize.STRING,
+          allowNull: false
```
Comment (jonnyalexbh) [inline migrations/migrations/20210406205811-name-of-migration.js]: please verify that the timestamps columns are being created `created_at` & `updated_at`

File: migrations/migrations/20210406205811-name-of-migration.js [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+    */
+      return queryInterface.createTable('user', {
+        id: {
+          type: Sequelize.INTEGER,
+          unique: true,
+          autoIncrement: true,
+          primaryKey: true,
+          allowNull: false
+        },
+        firstName: {
```
Comment (jonnyalexbh) [inline migrations/migrations/20210406205811-name-of-migration.js]: migration attributes is not camelCase are `snake_case`

File: app/models/user.js [hunk 12]
```diff
@@ -0,0 +1,32 @@
+module.exports = (sequelize, DataTypes) => {
+  const User = sequelize.define('user', {
```
Comment (jonnyalexbh) [inline app/models/user.js]: validate model indentation with `npm run lint`

File: migrations/migrations/20210406205811-name-of-migration.js [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+          type: Sequelize.STRING,
+          unique: true,
+          allowNull: false
+        },
+        password: {
+          type: Sequelize.STRING,
+          allowNull: false
+        }
+      }, {
+        timestamps: true
```
Comment (jonnyalexbh) [inline migrations/migrations/20210406205811-name-of-migration.js]: I think this property ` timestamps: true` is not correct in the migration this goes in the model

File: migrations/migrations/20210406205811-name-of-migration.js (hunk 9 above)
Comment (DanielAugustoUruena) [reply migrations/migrations/20210406205811-name-of-migration.js]: table name fixed

File: migrations/migrations/20210406205811-name-of-migration.js (hunk 8 above)
Comment (DanielAugustoUruena) [reply migrations/migrations/20210406205811-name-of-migration.js]: table name fixed

File: migrations/migrations/20210406205811-name-of-migration.js (hunk 10 above)
Comment (DanielAugustoUruena) [reply migrations/migrations/20210406205811-name-of-migration.js]: timestamp fields added

File: migrations/migrations/20210406205811-name-of-migration.js (hunk 11 above)
Comment (DanielAugustoUruena) [reply migrations/migrations/20210406205811-name-of-migration.js]: column name fixed

File: app/models/user.js (hunk 12 above)
Comment (DanielAugustoUruena) [reply app/models/user.js]: indentation fixed

File: migrations/migrations/20210406205811-name-of-migration.js (hunk 13 above)
Comment (DanielAugustoUruena) [reply migrations/migrations/20210406205811-name-of-migration.js]: property removed

**Review by jonnyalexbh — APPROVED** [review]
💯 Good job

**Pull request merged**